### PR TITLE
Added error handling to multiple functions in R/assign_job_queue.R, R/blastWrappers.R

### PR DESCRIPTION
**Pull request by Seyi007**
- Implement error handling for mapOption2Process, get_proc_medians, write_proc_medians_table, get_proc_weights, advanced_opts2est_walltime, assign_job_queue, and plot_estimated_walltimes .
- Validate input arguments for each function to ensure they meet expected criteria.
- Use tryCatch blocks to handle errors and warnings gracefully.
- Provide informative error messages and detailed logging where appropriate.
- Ensure functions fail gracefully and provide useful feedback.

- [x] Feature (adds or updates new capabilities)
- [ ] Bug fix (fixes an issue).
- [x] Enhancement (adds functionality).
- [ ] Breaking change (these changes would cause existing functionality to not work as expected).

# Checklist

- [x] I have read and followed the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] I have searched for existing content to ensure this is not a duplicate.
- [x] I have performed a self-review of these additions (including spelling, grammar, and related).
- [x] I have added comments to my code to help provide understanding.
- [ ] I have added a test which covers the code changes found within this PR.
- [x] I have deleted all non-relevant text in this pull request template.
- [x] **Reviewer assignment**: @jananiravi @the-mayer

### CODE REVIEW

**Review by the-mayer — CHANGES_REQUESTED** [review]
Thanks for the extra work here @Seyi007! Providing error handling for these functions will definitely be a worthwhile endeavor. I've rebased this PR on the current main branch to pull in some function name changes and .Rd files that have been submitted in other PRs.

I'd like to see a few modifications before I review further:

## `tryCatch()`
In R, the `finally` portion is _always_ evaluated, even if the function generates an error or warning. 

Printing “{function} execution completed.” might introduce confusion among our users, as they may think the function executed successfully even if an error occurred. Additionally, it could be overly verbose. I’d prefer to reserve the finally portion of each tryCatch() for cleanup or custom handler functions rather than messaging.

## `print()` VS `message()`
This is both a stylistic and functional request. Given the choice between `message()` and `print()` I'd prefer that we use `message()`. Essentially, when using message, users may choose to suppress the additional messaging with `suppressWarnings()` or `suppressMessages()` -- `print()` is harder to escape and I'd rather give the user the choice.

## Best option 
Use `rlang::inform()`, `rlang::warn()`, and `rlang::abort()` where appropriate: 
- https://rlang.r-lib.org/reference/abort.html

## Next Steps
Please consider the suggested modifications and submit the PR again for further review. Your efforts are greatly appreciated, and I’m looking forward to seeing the improvements!

**Comment by jananiravi** [comment]
Thanks for all your work on this @Seyi007. excellent points, @the-mayer! Looking forward to the changes!

**Comment by Seyi007** [comment]
Duly noted @the-mayer , I'll get to work on this!

**Review by falquaddoomi — COMMENTED** [review]
Nice work! Generally, I think having better error reporting will really help things, so thank you for the contribution.

I'm inexperienced at R, so take my comments with a grain of salt. Anyway, I made some suggestions in keeping with what @the-mayer had recommended in a previous comment.

File: R/blastWrappers.R [hunk 1]
```diff
@@ -55,20 +86,48 @@ run_deltablast <- function(deltablast_path, db_search_path,
...
+      )
     )
     print(Sys.time() - start)
+  }, error = function(e) {
+    message(paste("Error in run_rpsblast: ", e))
+  }, warning = function(w) {
+    message(paste("Warning in run_rpsblast: ", w))
+  }, finally = {
+    message("run_rpsblast completed")
+  })
```
[the remark this answers is not in the document]
Comment (Seyi007) [reply R/blastWrappers.R:131]: Yeah, I haven't updated the R/blastWrappers.R file with the suggestions from @the-mayer . I've only updated the first 2 files, and I'm currently working on the said file.

File: R/assign_job_queue.R [hunk 2]
```diff
@@ -1,3 +1,4 @@
+suppressPackageStartupMessages(library(rlang))
```
[the remark this answers is not in the document]
Comment (Seyi007) [reply R/assign_job_queue.R:1]: Duly noted! I'll include it in my next commit.

[the remark this answers is not in the document]
Comment (jananiravi) [reply R/assign_job_queue.R:1]: Also, if you are importing multiple functions from that library, it is better to include it as a dependency in the `imports` list [here](https://github.com/JRaviLab/MolEvolvR/blob/main/DESCRIPTION).

File: R/acc2lin.R [hunk 3]
```diff
@@ -87,27 +149,66 @@ addLineage <- function(df, acc_col = "AccNum", assembly_path,
...
+    # Attempt to process IPG to lineages
+    lins <- IPG2Lineage(accessions, ipgout_path,
+                        assembly_path, lineagelookup_path)
+  }, error = function(e) {
+    abort(
+      message = paste("An error occurred during IPG fetching
+                      or lineage processing:", e$message),
+      class = "lineage_processing_error",
+      # capturing the call stack
+      call = sys.call(),
```
[the remark this answers is not in the document]
Comment (jananiravi) [reply R/acc2lin.R:175]: Faisal, what's FAICT?

File: R/acc2lin.R [hunk 4]
```diff
@@ -194,17 +340,77 @@ efetchIPG <- function(accnums, out_path, plan = "sequential", ...) {
...
+  }, warning = function(w) {
+    warn(
+      message = paste("Warning: ", w$message),
+      class = "processing_warning",
+      call = sys.call(),
+      accessions = accessions,
+      ipg_file = ipg_file,
+      assembly_path = assembly_path,
+      lineagelookup_path = lineagelookup_path
+    )
```
[the remark this answers is not in the document]
Comment (Seyi007) [reply R/acc2lin.R:411]: Duly Noted!

**Comment by Seyi007** [comment]
Hi @jananiravi @falquaddoomi, I've implemented all the changes required from the previous review in my latest commit. Could you review this and let me know if I'm missing anything?

**Review by falquaddoomi — APPROVED** [review]
Nice, looks like you got to everything I raised.

**Comment by jananiravi** [comment]
@falquaddoomi @the-mayer followed the conversation but not the most recent line-by-line edits. Feel free to merge if this looks good. 

Thanks, @Seyi007 !

**Comment by Seyi007** [comment]
Yay! Thank you @falquaddoomi @jananiravi

**Review by the-mayer — APPROVED** [review]
Thanks @Seyi007, sending to main!

**Pull request merged**